crates/polars-core/src/chunked_array/mod.rs

```diff
@@ -442,6 +442,35 @@ impl AsSinglePtr for BinaryChunked {}
 #[cfg(feature = "object")]
 impl<T: PolarsObject> AsSinglePtr for ObjectChunked<T> {}
 
+pub enum ChunkedArrayLayout<'a, T: PolarsDataType> {
+    SingleNoNull(&'a T::Array),
+    Single(&'a T::Array),
+    MultiNoNull(&'a ChunkedArray<T>),
+    Multi(&'a ChunkedArray<T>),
+}
+
+impl<T> ChunkedArray<T>
+where
+    T: PolarsDataType,
+{
+    pub fn layout(&self) -> ChunkedArrayLayout<'_, T> {
```

Comment:
Are we going to use this?

Reply:
It was part of more refactors on the way that I got stuck on because of the issues with some types having more to it than their physical representation (e.g. Decimal).

I think it would be good to merge this anyway and start using it, I think it's cleaner than manually checking all over the place.